server/src/main/java/org/elasticsearch/common/logging/HeaderWarning.java

```diff
@@ -39,7 +41,7 @@ public class HeaderWarning {
         "Elasticsearch-" + // warn agent
         "\\d+\\.\\d+\\.\\d+(?:-(?:alpha|beta|rc)\\d+)?(?:-SNAPSHOT)?-" + // warn agent
         "(?:[a-f0-9]{7}(?:[a-f0-9]{33})?|unknown) " + // warn agent
-        "\"((?:\t| |!|[\\x23-\\x5B]|[\\x5D-\\x7E]|[\\x80-\\xFF]|\\\\|\\\\\")*)\"( " + // quoted warning value, captured
+        "\"(?<quotedStringValue>.*)\"( " + // quoted warning value, captured
```

Comment:
Please add comment to not allow other greedy qualifiers later to avoid excessive backtracking.